ContinuousRegistration/Source/make_hyperopt.py

```diff
@@ -0,0 +1,220 @@
+from ContinuousRegistration.Source.make_registration_scripts import parser, run as make_registration_scripts
+from ContinuousRegistration.Source.make_evaluation import run as make_evaluation
+from ContinuousRegistration.Source.util import write_json
+
+from hyperopt import fmin, tpe, hp
+from ContinuousRegistration.Source.datasets import load_datasets
+import shutil, numpy as np
+
+import sys, os, time
+from subprocess import Popen, check_output
+import glob, copy
+
+def exec_commands(parameters, cmds):
+    ''' Exec commands in parallel in multiple process
+    (as much as we have CPU)
+    '''
+    if not cmds: return # empty list
+
+    def done(p):
+        return p.poll() is not None
+    def success(p):
+        return p.returncode == 0
+    def fail():
+        sys.exit(1)
+
+    processes = []
+    while True:
+        while cmds and len(processes) < parameters.jobs:
+            task = cmds.pop()
+            print('Running %s ...' % task)
+            processes.append(Popen(task))
+
+        for p in processes:
+            if done(p):
+                if success(p):
+                    processes.remove(p)
+                else:
+                    fail()
+
+        if not processes and not cmds:
+            break
+        else:
+            time.sleep(1.0)
+
+class objective():
+    def __init__(self, parameters):
+        self.parameters = parameters
+
+        # Preprocess data
+        self.datasets = load_datasets(self.parameters)
+
+    def __call__(self, hyperopts):
+        # Write new blueprint to disk
+        blueprint = copy.deepcopy(parameters.blueprint)
+
+        for key, value in hyperopts.items():
+            blueprint["Components"].append(value)
+
+        write_json(os.path.join(parameters.submissions_directory, parameters.team_name, self.parameters.blueprint_file_name[0]), blueprint)
+
+        # Make shell scripts for running registrations. Scripts from previous iterations will be overridden
+        make_registration_scripts(self.parameters)
+
+        # Run registrations
+        if sys.platform == 'win32':
```

Comment:
since the exception is just printed you can move the `if/else` inside `try/except`